runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -180,6 +179,7 @@
                 this, proxy, updateEntry.getSMRMethod(),
                 updateEntry.getSMRArguments(), conflictObjects);
 
+        modifiedProxies.add(proxy);
         return addToWriteSet(proxy, updateEntry, conflictObjects);
```

Comment:
If you look at the implementation of `addToWriteSet`, it does a thread synchronization on root context:
```java
synchronized (getRootContext().getTransactionID()) {
        ....
}
```
Which makes me wonder if `modifiedProxies.add(proxy);` is sufficient. Also if you look at how we deal with nested transactions: 
```java
public void addTransaction(AbstractTransactionalContext tc) {
        ......
        mergeWriteSetInto(tc.getWriteSetInfo());
        ......
}
```
It merges writeSet, but the modifiedProxies does not do that. It might lose the tagging information. So maybe we should figure out a way to reuse `writeSet` for tagging or solve the above two issues.